derive_client/_clients/rest/http/transactions.py

```diff
@@ -0,0 +1,151 @@
+"""Transaction operations."""
+
+from decimal import Decimal
+from typing import Optional
+
+from derive_action_signing.module_data import DepositModuleData, WithdrawModuleData
+
+from derive_client.constants import CURRENCY_DECIMALS, INT64_MAX
+from derive_client.data.generated.models import (
+    MarginType,
+    PrivateDepositParamsSchema,
+    PrivateDepositResultSchema,
+    PrivateWithdrawParamsSchema,
+    PrivateWithdrawResultSchema,
+    PublicGetTransactionParamsSchema,
+    PublicGetTransactionResultSchema,
+)
+from derive_client.data_types import Currency
+
+
+class TransactionOperations:
+    """High-level transaction operations."""
+
+    def __init__(self, client):
+        """
+        Initialize transaction operations.
+
+        Args:
+            client: HTTPClient instance providing access to public/private APIs
+        """
+        self._client = client
+
+    def get(self, transaction_id: str) -> PublicGetTransactionResultSchema:
+        """Get a transaction by its transaction id."""
+
+        params = PublicGetTransactionParamsSchema(transaction_id=transaction_id)
+        response = self._client.public.get_transaction(params)
+        return response.result
+
+    def deposit_to_subaccount(
+        self,
+        amount: Decimal,
+        asset_name: str,
+        nonce: Optional[int] = None,
+        signature_expiry_sec: int = INT64_MAX,
+        is_atomic_signing: bool = False,
+    ) -> PrivateDepositResultSchema:
+        """Deposit from LightAccount smart contract wallet into subaccount."""
+
+        subaccount_id = self._client.subaccount_id
+        module_address = self._client._config.contracts.DEPOSIT_MODULE
+
+        currency = self._client.markets.get_currency(asset_name)
+        subaccount = self._client.account.get_subaccount()
```

Comment:
the subaccount query is no longer necessary per #106 caching this immutable state. The currency address however, which is required for action signing, is not cached and thus imposes an additional network call currently